src/edu/csus/ecs/pc2/core/report/ResultsCompareReport.java

```diff
@@ -0,0 +1,369 @@
+            }
+        }
+        lines.add("  Comparision summary " + compSummary);
+
+        String previousKey = "";
+        int foundCount = 0;
+
+        if (differentFields != 0) {
+
+            lines.add("");
+            
+            Collections.sort(fields, new FieldCompareRecordComparator());
+
+            for (FieldCompareRecord fieldCompareRecord : fields) {
+
+                String theKey = fieldCompareRecord.getKey();
+                if (!previousKey.equals(theKey)) {
+                    previousKey = theKey;
+                    foundCount = 0;
+                }
+
+                if (!fieldCompareRecord.getState().equals(ComparisonState.SAME)) {
+                    
+                    foundCount ++;
+                    if (foundCount == 1) {
+                        lines.add("");
+                        lines.add("Row key: " + theKey);
+                    }
+
+                    String stateString = fieldCompareRecord.getState().toString();
+                    
+                    String formattedLine = String.format("   field %-18s %8s '%s' '%s'", fieldCompareRecord.getFieldName(), stateString, fieldCompareRecord.getValueOne(),
+                            fieldCompareRecord.getValueTwo());
+
+                    switch (fieldCompareRecord.getState()) {
+                        case MISSING_SOURCE:
+                            stateString = getMissingSourceMessage();
+                            formattedLine = String.format("   field %-18s %8s '%s'", fieldCompareRecord.getFieldName(), stateString, fieldCompareRecord.getValueTwo());
+                            break;
+                        case MISSING_TARGET:
+                            stateString = getMissingTargetMessage();
+                            formattedLine = String.format("   field %-18s %8s '%s'", fieldCompareRecord.getFieldName(), stateString, fieldCompareRecord.getValueOne());
+                            break;
+                        case NOT_SAME:
+                            stateString = "Not identical";
+                            formattedLine = String.format("   field %-18s %8s '%s' '%s'", fieldCompareRecord.getFieldName(), stateString, fieldCompareRecord.getValueOne(),
+                                    fieldCompareRecord.getValueTwo());
+                            break;
+                        default:
+                            break;
+                    }
+
+                    lines.add(formattedLine);
+                }
+            }
+        }
+
+        return lines;
+    }
+    
+    @Override
+    public String[] createReport(Filter filter) {
+
+        String sourceDir = pc2ResultsDir;
+        String targetDir = primaryCCSResultsDir;
+        if (targetDir == null) {
+            throw new RuntimeException("Primary CCS directory is not assigned (is null)");
+        }
+
+        List<String> outList = new ArrayList<String>();
+
+        if (new File(targetDir).isDirectory()) {
```

Comment:
This `if` should also verify that `sourceDir` is a directory.  Otherwise, the calls to `FileComparisonUtitilies` methods a few lines below can fail because those methods _assume_ the `sourceDir` IS a directory (they append `File.separator` + `<filename>` to the specified "directory name", and then call `Utilities.loadFile(filename)`, which will fail and return an empty String[].   The result will be a failure report for a reason that's really not going to be obvious to the user.

I consider this a SHOW STOPPER that needs to be fixed before I would Approve the PR.

Reply:
I would suggest BLOCKER or BUG rather than SHOW STOPPER because a contest can be completed with accurate results and will not be impacted by this bug.   SHOW STOPPER is too much like a On Deck or cause for a contest failure.

Reply:
Ok, call it what you want.  I simply used the term "SHOW STOPPER" because, for me it needs to be fixed and that was the term we were using the last time we had a discussion about how to separate "nice if this were fixed" from "this needs to be fixed in order to gain my approval".

So the question is, are you going to fix it?

Reply:
That is an inappropriate question for a code review.

Reply:
focus on the code.

Reply:
Ok, I've deleted the question, although I'm not quite sure why you think it was inappropriate since it _directly_ bore on the question of the status of the PR.  Sorry if it somehow offended you.  I'll await your followup to the things in the PR which I requested changes on.